controllers/clustergroupupgrade_controller.go

```diff
@@ -248,152 +291,132 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 			}
 
 		}
-	} else if (backupCondition == nil || (backupCondition != nil && (backupCondition.Reason == string(utils.ConditionReasons.Completed) || backupCondition.Reason == string(utils.ConditionReasons.PartiallyDone)))) &&
-		(precachingCondition == nil || (precachingCondition != nil && precachingCondition.Reason == string(utils.ConditionReasons.Completed))) {
-		// All conditions met to begin the upgrade
+	} else if progressingCondition == nil || progressingCondition.Status == metav1.ConditionFalse {
 
-		progressingCondition := meta.FindStatusCondition(clusterGroupUpgrade.Status.Conditions, string(utils.ConditionTypes.Progressing))
+		if !*clusterGroupUpgrade.Spec.Enable {
+			utils.SetStatusCondition(
+				&clusterGroupUpgrade.Status.Conditions,
+				utils.ConditionTypes.Progressing,
+				utils.ConditionReasons.NotEnabled,
+				metav1.ConditionFalse,
+				"The ClusterGroupUpgrade CR is not enabled",
+			)
+			nextReconcile = requeueWithLongInterval()
+			r.updateStatus(ctx, clusterGroupUpgrade)
+			return
+		}
 
-		// Set the progressing condition if it is not present
-		if progressingCondition == nil || progressingCondition.Status == metav1.ConditionFalse {
-			var allManagedPoliciesExist bool
-			var managedPoliciesInfo policiesInfo
-			allManagedPoliciesExist, managedPoliciesInfo, err =
-				r.doManagedPoliciesExist(ctx, clusterGroupUpgrade, true)
-			if err != nil {
-				return
-			}
-			if allManagedPoliciesExist {
-				if !meta.IsStatusConditionTrue(clusterGroupUpgrade.Status.Conditions, string(utils.ConditionTypes.Validated)) {
-					utils.SetStatusCondition(
-						&clusterGroupUpgrade.Status.Conditions,
-						utils.ConditionTypes.Validated,
-						utils.ConditionReasons.Completed,
-						metav1.ConditionTrue,
-						"All managed policies exist",
-					)
-				}
+		// create backup
+		err = r.reconcileBackup(ctx, clusterGroupUpgrade)
+		if err != nil {
+			r.Log.Error(err, "reconcileBackup error")
+			return
+		}
 
-				// Build the upgrade batches.
-				err = r.buildRemediationPlan(ctx, clusterGroupUpgrade, managedPoliciesInfo.presentPolicies)
-				if err != nil {
+		backupCondition := meta.FindStatusCondition(clusterGroupUpgrade.Status.Conditions, string(utils.ConditionTypes.BackupSuceeded))
+		if backupCondition != nil && backupCondition.Status == metav1.ConditionFalse {
+			utils.SetStatusCondition(
+				&clusterGroupUpgrade.Status.Conditions,
+				utils.ConditionTypes.Progressing,
+				utils.ConditionReasons.Enabled,
```

Comment:
Hi @jc-rh, @fontivan, when backup is going on, although although enabled is true, progressing condition waits for backup to succeed, in that case, what could be an appropriate utils.ConditionReasons? For now, i kept it is enabled, which doesn't make a lot of sense. may be "NotStarted"? wdyt?

Reply:
I'm not sure you need to set a condition here. There will already be a BackupSucceeded/InProgress condition present while the backup is ongoing. Couldn't you just remove the check for backup condition and handle it inside the if condition?

Reply:
" Couldn't you just remove the check for backup condition and handle it inside the if condition?" --> are you suggesting it to move at line 330?

Reply:
oh ok I see what you mean now. Yeah this is probably fine then